### PR TITLE
#439 Sonarqube debt from Sprint 6,7 fixed

### PR DESCRIPTION
**Pull request by JeremiahTee**
This brings the debt to 0
Addressed all code smells, security vulnerabilities and bugs

No need to do this for Sprint 8,9 as we have Sonarcloud now (unless code smells get merged unchecked)

[Link to report](https://databoom.atlassian.net/l/c/axMLPoqa)

### CODE REVIEW

File: Server/src/services/authenticationService.ts [hunk 1]
```diff
@@ -174,14 +174,14 @@ export class AuthenticationService {
...
-  private async generateJwtToken(jwtPayload: IJwtParams): Promise<string> {
+  private generateJwtToken(jwtPayload: IJwtParams): string {
 
     let token: string;
     let tokenInformation;
     let jwtExpiry: number = 3000;
     const jwtAccessKey = process.env.ACCESS_SECRET_KEY;
 
-    token = await jwt.sign({ jwtPayload }, jwtAccessKey, {
+    token = jwt.sign({ jwtPayload }, jwtAccessKey, {
```
Comment (ds161833) [inline Server/src/services/authenticationService.ts:184]: jwt.sign returns a promise, this call should be awaited, why did you remove await?

File: Server/src/services/authenticationService.ts [hunk 2]
```diff
@@ -123,7 +123,7 @@ export class AuthenticationService {
       let jwtParams: IJwtParams;
       try {
         jwtParams = await AuthenticationModel.obtainJWTParams(forgotPasswordInfo.email);
-        accessToken = await this.generateJwtToken(jwtParams);
+        accessToken = this.generateJwtToken(jwtParams);
```
Comment (ds161833) [inline Server/src/services/authenticationService.ts:126]: same as the comment below

**Review by laithAwad — APPROVED** [review]
good work!

File: Server/src/services/authenticationService.ts (hunk 1 above)
Comment (JeremiahTee) [reply Server/src/services/authenticationService.ts:184]: sign method is actually synchronous apparently, sonarqube flagged it and checked the code
here's part of the description from source code
"Synchronously sign the given payload into a JSON Web Token string payload "

Comment (ds161833) [reply Server/src/services/authenticationService.ts:184]: can you attach a screenshot? its pretty weird, im surprised how sonarCube knows something extra we cant tell ourselves

Comment (ds161833) [reply Server/src/services/authenticationService.ts:184]: wait, weird intellisense told me it returns a promise before, now it says string, resolved 👍

**Review by ds161833 — APPROVED** [review]

File: Server/src/services/authenticationService.ts (hunk 1 above)
Comment (JeremiahTee) [reply Server/src/services/authenticationService.ts:184]: sorry I don't have the screenshot from sonarqube as i already marked the code smell as fixed, but the sign we are using in the code is the synchronous one without callback 
image

Comment (JeremiahTee) [reply Server/src/services/authenticationService.ts:184]: image

Comment (JeremiahTee) [reply Server/src/services/authenticationService.ts:184]: > wait, weird intellisense told me it returns a promise before, now it says string, resolved 👍

sounds good, thank you for double checking!
